staging/src/k8s.io/kubectl/pkg/cmd/auth/token.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// TokenOptions is the data required to perform a token request operation.
+type TokenOptions struct {
+	// ServiceAccountName is the name of the service account to request a token for. Required.
+	ServiceAccountName string
+	// NamespaceName is the namespace of the service account to request a token for. Required.
+	NamespaceName string
+
+	// BoundObjectKind is the kind of object to bind the token to. Optional. Can be Pod or Secret.
+	BoundObjectKind string
+	// BoundObjectName is the name of the object to bind the token to. Required if BoundObjectKind is set.
+	BoundObjectName string
+	// BoundObjectUID is the uid of the object to bind the token to. If unset, defaults to the current uid of the bound object.
+	BoundObjectUID string
+
+	// Audiences indicate the valid audiences for the requested token. If unset, defaults to the Kubernetes API server audiences.
+	Audiences []string
+
+	// ExpirationSeconds is the requested token lifetime. Optional.
+	ExpirationSeconds int64
+
+	// CoreClient is the API client used to request the token. Required.
+	CoreClient corev1client.CoreV1Interface
+
+	// IOStreams are the output streams for the operation. Required.
+	genericclioptions.IOStreams
+}
+
+var (
+	tokenLong = templates.LongDesc(`Request a service account token.`)
+
+	tokenExample = templates.Examples(`
+		# Request a token to authenticate to the kube-apiserver as myapp in the current namespace
+		kubectl auth token --service-account myapp
```

Comment:
in `kubectl create rolebinding`, I think this is `--serviceaccount=`

Reply:
not relevant after relocating to `kubectl create token $SERVICE_ACCOUNT_NAME`